react/features/toolbox/components/AudioRouteButton.js

```diff
@@ -0,0 +1,153 @@
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * TODO.
+         */
+        dispatch: PropTypes.func,
+
+        /**
+         * The name of the Icon of this {@code AudioRouteButton}.
+         */
+        iconName: PropTypes.string,
+
+        /**
+         * The style of the Icon of this {@code AudioRouteButton}.
+         */
+        iconStyle: PropTypes.object,
+
+        /**
+         * {@code AudioRouteButton} styles.
+         */
+        style:
+            PropTypes.oneOfType([
+                PropTypes.array,
+                PropTypes.object
+            ]),
+
+        /**
+         * The color underlying the button.
+         */
+        underlayColor: PropTypes.any
+    };
+
+    _volumeComponent: ?Object;
+
+    /**
+     * Initializes a new {@code AudioRouteButton} instance.
+     *
+     * @param {Object} props - The React {@code Component} props to initialize
+     * the new {@code AudioRouteButton} instance with.
+     */
+    constructor(props) {
+        super(props);
+
+        /**
+         * The internal reference to the React {@code MPVolumeView} for
+         * showing the volume control view.
+         *
+         * @private
+         * @type {ReactComponent}
+         */
+        this._volumeComponent = null;
+
+        // Bind event handlers so they are only bound once per instance.
+        this._onClick = this._onClick.bind(this);
+        this._setVolumeComponent = this._setVolumeComponent.bind(this);
+    }
+
+    _onClick: () => void;
+
+    /**
+     * Handles clicking/pressing this {@code AudioRouteButton} by showing an
+     * audio route picker.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onClick() {
+        if (Platform.OS === 'ios') {
+            const handle = findNodeHandle(this._volumeComponent);
+
+            NativeModules.MPVolumeViewManager.show(handle);
+        } else if (Platform.OS === 'android') {
+            this.props.dispatch(openDialog(AudioRoutePickerDialog));
+        }
+    }
+
+    _setVolumeComponent: (?Object) => void;
+
+    /**
+     * Sets the internal reference to the React Component wrapping the
+     * {@code MPVolumeView} component.
+     *
+     * @param {ReactComponent} component - React Component.
+     * @returns {void}
+     */
+    _setVolumeComponent(component) {
+        this._volumeComponent = component;
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { iconName, iconStyle, style, underlayColor } = this.props;
+
+        return (
+            <View>
+                <ToolbarButton
+                    iconName = { iconName }
+                    iconStyle = { iconStyle }
+                    onClick = { this._onClick }
+                    style = { style }
+                    underlayColor = { underlayColor } />
+                <MPVolumeView
```

Comment:
I think that's too complicated for its own good. It should be binary and the binary nature should be expressed as clearly as possible everywhere: either you're using/you have `MPVolumeView` or you're using/you have `AudioRoutePickerDialog`.

Reply:
Ok.